src/components/Products.jsx

```diff
@@ -0,0 +1,131 @@
+import { fetchProduct } from "../store/actions/productActions";
+import { titleSelector, categorySelector, priceSelector, ratingSelector, stockSelector, totalSelector } from "../store/selectors/productSelector";
+import { listCategorySelector } from "../store/selectors/categorySelector";
+import Category from "./core/Category";
+import { useInView } from "react-intersection-observer";
+import { fetchCategory } from "../store/actions/categoryAction";
+
+const Products = () => {
+  const dispatch = useDispatch();
+  const [page, setPage] = useState(0);
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const [prevCategory, setPrevCategory] = useState("");
+
+  // Per page/scroll fetch limit = 10-15
+  const limit = 10;
+
+  // Fetching Categories from Categories-only API 
+  const listCategory = useSelector(listCategorySelector);
+
+  const titles = useSelector(titleSelector);
+  const categories = useSelector(categorySelector);
+  const prices = useSelector(priceSelector);
+  const ratings = useSelector(ratingSelector);
+  const stocks = useSelector(stockSelector);
+  const loading = useSelector((state) => state.product.loading);
+  const error = useSelector((state) => state.product.error);
+  const total = useSelector(totalSelector);
+
+  useEffect(() => {
+    dispatch(fetchCategory());
+  }, [dispatch]);
+
+  useEffect(() => {
+    const shouldReplace = selectedCategory !== prevCategory;
+    if (selectedCategory !== prevCategory) {
+      setPage(0);
+      setPrevCategory(selectedCategory);
+    }
+    dispatch(fetchProduct(limit, page * limit, selectedCategory, shouldReplace));
+  }, [dispatch, page, selectedCategory, prevCategory]);
+
+  const handleCategoryClick = (category) => {
+    setSelectedCategory(category);
+  };
+
+  const { ref, inView } = useInView({
+    triggerOnce: false,
+    onChange: (inView) => {
+      if (inView && !loading && page * limit < total) {
+        setPage((prevPage) => prevPage + 1);
+      }
+    },
+  });
+
+  if (loading && page === 0) return <p>Loading...</p>;
+  if (error) return <p>Error: {error}</p>;
+
+  return (
+    <div className="p-4">
+      <h1 className="text-center font-bold text-2xl py-4">Product Table</h1>
+
+      <div className="p-2 flex items-center flex-wrap gap-2 font-semibold">
+        <div>Filter from Categories: </div>
+        {listCategory.length > 0
+          ? listCategory.slice(0, 5).map((category, index) => (
+            <Category
+              key={index}
+              name={category.name}
+              onClick={() => handleCategoryClick(category.slug)}
+            />
+          ))
+          : "No categories available"}
+        <button
+          className="bg-gray-500 hover:bg-gray-700 text-white font-bold py-2 px-4 rounded"
+          onClick={() => {
+            setSelectedCategory("");  
+            setPage(0);               
+            setPrevCategory("");      
+            dispatch(fetchProduct(10, 0, "", true)); //Force Clear
```

Comment:
use the **limit** variable declared at the top 
<img width="290" alt="image" src="https://github.com/user-attachments/assets/7bdea5db-f32c-4815-833d-c4ec363d88e0">

Reply:
`limit` variable reused. Value updated to `15`.
Resolved ✅